File: mock_interviews/leetcode/575.cpp
class Solution {
public:
    int distributeCandies(vector<int>& candies)
    {
        map<int, int> hashing;
        for (int i = 0; i < candies.size(); i++)
            if (hashing.find(candies[i]) == hashing.end())
                hashing[candies[i]] = 1;
            else
                hashing[candies[i]]++;
        if (hashing.size() > candies.size() / 2)
            return candies.size() / 2;
        else
            return hashing.size();
    }
};

File: mock_interviews/leetcode/575.py
class Solution:
    def distributeCandies(self, candies):
        """
        :type candies: List[int]
        :rtype: int
        """
        hashing = {}
        for candy in candies:
            if candy in hashing.keys():
                hashing[candy] += 1
            else:
                hashing[candy] = 0
        if len(hashing) > len(candies)/2:
            return int(len(candies)/2)
        else:
            return len(hashing)
